docs/user-guide/catalog.md

```diff
@@ -92,41 +96,117 @@ Enabling the *Show metadata template* option appears a text editor through witch
 
 <img src="../img/catalog/metadata-det.gif" class="ms-docimage"  style="max-width:500px;"/>
 
 !!! note
     If some metadata are missing, the server response will be `source Not Available`
+
+### WMS/WMTS Catalog
+
+WMS and WMTS Services are [OGC Standards](https://www.ogc.org/standards) protocol for publishing maps (and tile maps) on the Internet. The user can add these kind of services as catalogs to browse and add to the map the layers published using these protocols.
+
+In **General Settings** the user can set the title he wants to assign to this service and the URL of the service to configure the service.
+
+In **Advanced Settings** the user can set, other than the standard options, also:
+
+* *Format*: the default image format for the layers added to the map (`png`, `png8`, `jpeg`, `vnd.jpeg-png` or `gif`). Setting this is particularly useful when the user wants to use optimized formats by default (`png8`, `vnd.jpeg-png`) for all the layers, without having to select it for each layer in layer settings.
+
+### TMS Catalog
+
+TMS Catalog type includes some not official/not standard protocol for serving maps as tiles (i.e. splitting map up into a pyramid of images at multiple zoom levels).
+MapStore allows to add to the map the following services providers:
+
+* Custom TMS service, specifying the URL template for the tiles.
+* TMS 1.0.0 , setting the URL
+* Select from a list known TMS services, with all the variants.
+
+<img src="../img/catalog/tms.jpg" class="ms-docimage"  style="max-width:500px;"/>
+<p class="ms-doc-caption">Select provider for TMS. The list of providers contains "custom", "TMS 1.0.0" and other resources</p>
+
+!!! note
+   Because some of these services are not standard, using them in different CRS may cause problems. So please take into account that switching CRS may cause problem when these layers are on the map.
+
+#### Custom TMS
+
+Selecting the **custom** provider the user can insert the tile URL template manually. The URL template is an URL with some placeholder that will be replaced with variables. The placeholder are identified by strings between brackets. e.g.: `{variable_name}`.
+
+<img src="../img/catalog/custom_tms.jpg" class="ms-docimage"  style="max-width:500px;"/>
+<p class="ms-doc-caption">Edit a custom TMS</p>
+
+
+Allowed placeholder are:
+
+* `{x}`,`{y}`,`{z}`: coordinates of the tiles
+* `{s}`: subdomains, this provides support for *domain sharding*. By default this is `["a", "b", "c"]`. User can customize the default by adding options.subdomains.
+
+*example:*
+
+```json
+{
+  "options": {
+    "subdomains": ["a", "b", "c", "d", "e"]
+  }
+}
+```
+
+When the user saves this custom catalog service and clicking on search, he will see only one result, that can be added to the map. (In the future custom provider can provide also variants, where in advanced settings the user can customize some variables, to have multiple results for the given resource).
```

Comment:
When the user saves this custom catalog service and clicks on search, he will see only one result, that can be added on the map: variants are not currently sopported in MapStore for this provider type.